install_config/advanced_ldap_configuration/configuring_extended_ldap_attributes.adoc

```diff
@@ -0,0 +1,181 @@
+= Configuring Extended LDAP Attributes
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+This topic builds upon
+link:../advanced_ldap_configuration/ssd_for_ldap_failover.html[Setting up SSSD
+for LDAP Failover] and
+link:../advanced_ldap_configuration/configuring_form_based_authentication.html[Configuring
+Form-Based Authentication] and focuses on configuring extended Lightweight
+Directory Access Protocol (LDAP) attributes.
+
+[[configuring-extended-ldap-attributes-prerequisites]]
+== Prerequisites
+
+* SSSD 1.12.0 or later. This is available on Red Hat Enterprise Linux 7.0 and
+later.
+* mod_lookup_identity 0.9.4 or later.
+** This is not yet available on any version of Red Hat Enterprise Linux 7, but the
+Red Hat Package Managers (RPMs) for this platform are
+https://copr.fedorainfracloud.org/coprs/adelton/identity_demo/[available for
+upstream] until they arrive in Red Hat Enterprise Linux.
+
+[[configuring-extended-ldap-attributes-config-sssd]]
+== Configuring SSSD
+You need to ask System Security Services Daemon (SSSD) to look up attributes in
+LDAP that it normally does not care about for simple system-login use-cases. In
+the case of {product-title}, there is only one such attribute: email. So, you need to:
+
+. Modify the *[domain/DOMAINNAME]* section of *_/etc/sssd/sssd.conf_* on the
+authenticating proxy and add this attribute:
++
+----
+[domain/example.com]
+...
+ldap_user_extra_attrs = mail
+----
+
+. Tell SSSD that it is acceptable for this attribute to be retrieved by
+Apache. Add the following two lines to the *[ifp]* section of
+*_/etc/sssd/sssd.conf_*:
++
+====
+----
+[ifp]
+user_attributes = +mail
+allowed_uids = apache, root
+----
+====
+
+. Restart SSSD and test this configuration:
++
+====
+----
+# systemctl restart sssd.service
+
+# getent passwd <username>
+username:*:12345:12345:Example User:/home/username:/usr/bin/bash
+
+# gdbus call \
+        --system \
+        --dest org.freedesktop.sssd.infopipe \
+        --object-path /org/freedesktop/sssd/infopipe/Users/example_2ecom/12345 \
+        --method org.freedesktop.DBus.Properties.Get \
+        "org.freedesktop.sssd.infopipe.Users.User" "extraAttributes"
+(<{'mail': ['username@example.com']}>,)
+----
+====
```

Comment:
This `gdbus` invocation is pretty complicated for the average user and they probably won't know how to escape the domain (see: "example_2ecom") properly. Might be better to skip the test here and leave that as a debugging tool in a troubleshooting doc somewhere.